pkg/api/helpers/errors.go

```diff
@@ -108,25 +108,29 @@ func ResponseAsError(response interface{}) error {
 			Err:     ErrRequestFailed,
 		}
 	}
-
-	f := r.FieldByName("HTTPResponse")
-	if !f.IsValid() {
-		return fmt.Errorf("[no HTTPResponse]: %w", ErrRequestFailed)
-	}
-	httpResponse, ok := f.Interface().(*http.Response)
-	if !ok {
-		return fmt.Errorf("%w: no HTTPResponse", ErrRequestFailed)
+	var f reflect.Value
+	var httpResponse *http.Response
+	var ok bool
+	if reflect.TypeOf(response).String() != "*http.Response" {
```

Comment:
I would rather _not_ do this: now `ResponseAtError` can take `*http.Response` (from package `net/http`), or it can take some`*Response` type from code generated by the API.

The only reason for using reflection here is that there is no interface that generalizes the `*Response` types and lets us get the `*http.Response`; that's a bug in the API generator or a bug in Go not having generics.  But in order to handle passing a known `*http.Response`, you could simply add a new function `HTTPResponseAsError` -- it's this function starting at line 126 "` 	if httpResponse == nil || isOK(httpResponse.StatusCode) {`".  And the old `ResponseAsError` would simply end before that line with something like `return HTTPResponseAsError(httpResponse, err)`.